source/Handlebars/IO/HtmlEncoderLegacy.cs

```diff
@@ -0,0 +1,76 @@
+﻿using HandlebarsDotNet.StringUtils;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace HandlebarsDotNet
+{
+    /// <summary>
+    /// <inheritdoc />
+    /// Produces <c>HTML</c> safe output.
+    /// <para>
+    /// This will encode non-ascii characters.
+    /// this will not encode '=', '`' or ''' (single quote).
+    /// </para>
+    /// </summary>
+    public class HtmlEncoderLegacy : ITextEncoder
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void Encode(StringBuilder text, TextWriter target)
+        {
+            if (text == null || text.Length == 0) return;
+
+            EncodeImpl(new StringBuilderEnumerator(text), target);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void Encode(string text, TextWriter target)
+        {
+            if (string.IsNullOrEmpty(text)) return;
+
+            EncodeImpl(new StringEnumerator(text), target);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void Encode<T>(T text, TextWriter target) where T : IEnumerator<char>
+        {
+            if (text == null) return;
+
+            EncodeImpl(text, target);
+        }
+
+        private void EncodeImpl<T>(T text, TextWriter target) where T : IEnumerator<char>
```

Comment:
Why not `static`?

Reply:
Fixed. No reason, this was left from my first implementation where it did need to be instance method.